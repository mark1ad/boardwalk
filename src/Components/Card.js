// Card
//
// Displays a string.
//
// Attributes:
//  name: string, required.
//  id: string, required.
//  submit: callback for when card is clicked. required.

import React from 'react';
import { PropTypes } from 'prop-types';

const Card = (props) => {
  const submit = e => {
    e.preventDefault();
    props.submit();
  }

  return (
    <div className={props.class} onClick={submit}>
      {props.name}
    </div>
  )
}

Card.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  submit: PropTypes.func.isRequired
}

export default Card;
