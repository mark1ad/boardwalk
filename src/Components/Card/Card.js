// Card
//
// Displays a string.
//
// Attributes:
//  name: string, required.
//  id: string, required.
//  submit: callback for when card is clicked. required.

import React from "react";
import { PropTypes } from "prop-types";

/**
 * Displays a string and performs and action when pressed.
 * (Really just a button)
 */
const Card = (props) => {
  const submit = e => {
    e.preventDefault();
    props.submit();
  };

  return (
    <div className={props.class} onClick={submit}>
      {props.name}
    </div>
  );
};

Card.propTypes = {
  /** String to display */
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  /** Callback for submit */
  submit: PropTypes.func.isRequired,
  class: PropTypes.string.isRequired
};

export default Card;
