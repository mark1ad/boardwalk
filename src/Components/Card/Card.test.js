import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Card from './Card';

const mockSubmitFunction = jest.fn();

it('renders without crashing', () => {
  const id = 1;
  const submit = () => {};
  const div = document.createElement('div');
  ReactDom.render(<Card name="name" id={id} submit={submit} class="class" />, div);
})

it("renders correctly", () => {
  const id = 1;
  const submit = () => {};
  const tree = renderer
    .create(<Card name="name" id={id} submit={submit} class="class" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

it("should call the mock sumbit function", () => {
  const id = 1;
  const card = shallow(<Card name="name" id={id} submit={mockSubmitFunction} class="test-card" />);
  card.simulate('click', {preventDefault() {}})
  expect(mockSubmitFunction.mock.calls.length).toBe(1);
})
