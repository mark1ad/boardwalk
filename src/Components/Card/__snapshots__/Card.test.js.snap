// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="class"
  onClick={[Function]}
>
  name
</div>
`;
